src/components/ui/Form.tsx

```diff
@@ -0,0 +1,153 @@
+
+import { Root as LabelPrimitiveRoot } from '@radix-ui/react-label';
+import { Slot } from '@radix-ui/react-slot';
+import {
+  ComponentPropsWithoutRef,
+  createContext,
+  ElementRef,
+  forwardRef,
+  HTMLAttributes,
+  useContext,
+  useId,
+} from 'react';
+import {
+  Controller,
+  ControllerProps,
+  FieldPath,
+  FieldValues,
+  FormProvider,
+  useFormContext,
+} from 'react-hook-form';
+
+const Form = FormProvider;
+
+type FormFieldContextValue<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+> = {
+  name: TName;
+};
+
+const FormFieldContext = createContext<FormFieldContextValue>({} as FormFieldContextValue);
+
+const FormField = <
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+>({
+  ...props
+}: ControllerProps<TFieldValues, TName>) => {
+  return (
+    <FormFieldContext.Provider value={{ name: props.name }}>
+      <Controller {...props} />
+    </FormFieldContext.Provider>
+  );
+};
+
+const useFormField = () => {
+  const fieldContext = useContext(FormFieldContext);
+  const itemContext = useContext(FormItemContext);
+  const { getFieldState, formState } = useFormContext();
+
+  const fieldState = getFieldState(fieldContext.name, formState);
+
+  if (!fieldContext) {
+    throw new Error('useFormField should be used within <FormField>');
+  }
+
+  const { id } = itemContext;
+
+  return {
+    id,
+    name: fieldContext.name,
+    formItemId: `${id}-form-item`,
+    formMessageId: `${id}-form-item-message`,
+    ...fieldState,
+  };
+};
+
+type FormItemContextValue = {
+  id: string;
+};
+
+const FormItemContext = createContext<FormItemContextValue>({} as FormItemContextValue);
+
+const FormItem = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
+  ({ className, ...props }, ref) => {
+    const id = useId();
+
+    return (
+      <FormItemContext.Provider value={{ id }}>
+        <div ref={ref} className={className} {...props} />
+      </FormItemContext.Provider>
+    );
+  },
+);
+FormItem.displayName = 'FormItem';
+
+const FormLabel = forwardRef<
+  ElementRef<typeof LabelPrimitiveRoot>,
+  ComponentPropsWithoutRef<typeof LabelPrimitiveRoot> & {
+    required?: boolean;
+  }
+>(({ className, children, required, ...props }, ref) => {
+  const { formItemId } = useFormField();
+
+  return (
+    <Label ref={ref} className={cn('mb-2', className)} htmlFor={formItemId} {...props}>
+      {children}
+      {required && <span className="text-danger">*</span>}
+    </Label>
+  );
+});
+FormLabel.displayName = 'FormLabel';
+
+const FormControl = forwardRef<ElementRef<typeof Slot>, ComponentPropsWithoutRef<typeof Slot>>(
+  ({ ...props }, ref) => {
+    const { error, formItemId } = useFormField();
+
+    return (
+      <Slot
+        ref={ref}
+        id={formItemId}
+        // aria-describedby={!error ? `${formDescriptionId}` : `${formDescriptionId} ${formMessageId}`}
```

Comment:
Oh, I realized now that you removed `FormDescription` component from original file, so we don't need description id. My question NO.1 is solved now.
Please just share with me about No.2, the intention on commenting out on line 115

Reply:
I forgot to address the comment. Please let me research about ARIA first. I will update it after that.

Reply:
Thank you for reserching! I'll just leave my opinion below, please correct it or just ignore it on your convenience!
IMO, since we removed `FormDescription` from original file, we will not have description for our `Form` component, thus we don't need `aria-describedby` anymore.

Reply:
The conclusion was reached in reference to the following item: [`7e5b438` (#167)](https://github.com/genesis-tech-tribe/nishiki-frontend/pull/167/commits/7e5b438f7b7d0a87df995192160dc5a221e0bd51).